crates/ty_project/src/db.rs

```diff
@@ -128,18 +128,51 @@ impl ProjectDatabase {
     /// to the CLI after a typechecker run.
     pub fn salsa_memory_dump(&self) -> SalsaMemoryDump {
         let memory_usage = <dyn salsa::Database>::memory_usage(self);
-        let mut ingredients = memory_usage.structs;
-        let mut memos = memory_usage.queries.into_iter().collect::<Vec<_>>();
+        let mut ingredients = memory_usage
+            .structs
+            .into_iter()
+            .filter(|ingredient| ingredient.count() > 0)
+            .collect::<Vec<_>>();
+        let mut memos = memory_usage
+            .queries
+            .into_iter()
+            .filter(|(_, memos)| memos.count() > 0)
+            .collect::<Vec<_>>();
+
+        ingredients.sort_by_key(|ingredient| {
+            let heap_size = ingredient.heap_size_of_fields().unwrap_or_else(|| {
+                // Salsa currently does not expose a way to track the heap size of interned
+                // query arguments.
+                if ingredient.debug_name().contains("interned_arguments") {
+                    0
+                } else {
+                    panic!(
+                        "expected `heap_size` to be provided by Salsa struct `{}`",
+                        ingredient.debug_name()
+                    )
```

Comment:
I think I'd prefer a `tracing::warn` over a hard panic (with a fallback to 0). It makes me aware that the numbers are under reporting but it might not matter for the analysis I'm doing right now (so I don't have to fix it immediately)